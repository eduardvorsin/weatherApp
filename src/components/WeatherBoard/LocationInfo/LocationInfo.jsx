import PropTypes from 'prop-types';
import classes from './LocationInfo.module.css';

export default function LocationInfo({
  className,
  text,
  testId,
}) {
  return (
    <p
      className={`${classes['location-info']} ${className}`}
      data-testid={testId}
    >
      <span className={classes['location-info__text']}>
        {text}
      </span>
      <img
        className={classes['location-info__icon']}
        src="/assets/icons/location.svg"
        alt="location icon"
      />
    </p>
  );
}

LocationInfo.defaultProps = {
  className: '',
  text: '',
  testId: '',
};

LocationInfo.propTypes = {
  className: PropTypes.string,
  text: PropTypes.string,
  testId: PropTypes.string,
};
