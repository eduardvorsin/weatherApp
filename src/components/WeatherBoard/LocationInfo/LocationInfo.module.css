.location-info {
  margin: 0;
  display: flex;
  align-items: center;
  justify-content: center;
  color: var(--brand-color);
  max-width: 250px;
}

.location-info__text {
  font-weight: 700;
  font-size: 27px;
  margin-right: 5px;
}

.location-info__icon {
  width: 30px;
  height: 30px;
}

@media(min-width:577px) {
  .location-info__text {
    font-size: 35px;
  }

  .location-info__icon {
    width: 35px;
    height: 35px;
  }
}
