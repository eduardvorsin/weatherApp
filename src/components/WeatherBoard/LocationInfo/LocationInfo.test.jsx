import { render, screen } from '@testing-library/react';
import LocationInfo from './LocationInfo';

describe('LocationInfo component tests', () => {
  it('renders correctly', () => {
    render(<LocationInfo
      className="test-class"
      text="Лондон"
    />);

    expect(screen.getByText(/Лондон/)).toBeInTheDocument();
  });

  it('renders without props', () => {
    render(<LocationInfo
      testId="location-info"
    />);

    expect(screen.getByTestId('location-info')).toBeInTheDocument();
  });

  it('the classname is correctly assigned to the element', () => {
    render(<LocationInfo
      className="test-class"
      testId="location-info"
    />);

    expect(screen.getByTestId('location-info')).toHaveClass('test-class');
  });

  it('snapshot with default data', () => {
    render(<LocationInfo
      testId="location-info"
    />);

    expect(screen.getByTestId('location-info')).toMatchSnapshot();
  });

  it('snapshot with mock text', () => {
    render(<LocationInfo
      className="test-class"
      text="Лондон"
      testId="location-info"
    />);

    expect(screen.getByTestId('location-info')).toMatchSnapshot();
  });
});
