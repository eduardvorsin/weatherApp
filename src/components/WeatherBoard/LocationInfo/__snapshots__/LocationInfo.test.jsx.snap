// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationInfo component tests snapshot with default data 1`] = `
<p
  class="location-info "
  data-testid="location-info"
>
  <span
    class="location-info__text"
  />
  <img
    alt="location icon"
    class="location-info__icon"
    src="/assets/icons/location.svg"
  />
</p>
`;

exports[`LocationInfo component tests snapshot with mock text 1`] = `
<p
  class="location-info test-class"
  data-testid="location-info"
>
  <span
    class="location-info__text"
  >
    Лондон
  </span>
  <img
    alt="location icon"
    class="location-info__icon"
    src="/assets/icons/location.svg"
  />
</p>
`;
